Задача 3. Проведите тест гипотезы. Продавец утверждает, что средний вес пачки печенья
составляет 200 г.
Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
Известно, что их веса распределены нормально.
Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна
99%? (Провести двусторонний тест.)

In [29]:
arr = [202, 203, 199, 197, 195, 201, 200, 204, 194, 190]
avr = sum(arr)/len(arr)
M_x = 200
a=0.01
n = 10
avr


198.5

In [23]:
import numpy as np
from scipy import stats




In [33]:

arr_v = np.var(arr,ddof=1)

t1 = stats.t.ppf(a/2,df=2*(n-1))
t2 = stats.t.ppf(1-a/2,df=2*(n-1))
t_emp =  (avr-M_x)/(np.sqrt(arr_v/n))

t1,t2,t_emp



(-2.8784404727135864, 2.878440472713585, -1.0651074037450896)

Вывод

Нулевая гипотеза, что средней вес пачки равен 200 г не отвергается (т.к. значение лежит м/у -2.878 и 2.878)